src/sql/workbench/contrib/notebook/test/browser/MarkdownTextTransformer.test.ts

```diff
@@ -0,0 +1,200 @@
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as TypeMoq from 'typemoq';
+import * as assert from 'assert';
+import * as dom from 'vs/base/browser/dom';
+
+import { MarkdownTextTransformer, MarkdownButtonType } from 'sql/workbench/contrib/notebook/browser/markdownToolbarActions';
+import { NotebookService } from 'sql/workbench/services/notebook/browser/notebookServiceImpl';
+import { TestInstantiationService } from 'vs/platform/instantiation/test/common/instantiationServiceMock';
+import { TestLifecycleService, TestEnvironmentService, TestAccessibilityService, TestTextResourceConfigurationService, TestEditorGroupsService, TestEditorService } from 'vs/workbench/test/browser/workbenchTestServices';
+import { INotebookService } from 'sql/workbench/services/notebook/browser/notebookService';
+import { CellModel } from 'sql/workbench/services/notebook/browser/models/cell';
+import { MockContextKeyService } from 'vs/platform/keybinding/test/common/mockKeybindingService';
+import { TestThemeService } from 'vs/platform/theme/test/common/testThemeService';
+import { TextModel } from 'vs/editor/common/model/textModel';
+import { isUndefinedOrNull } from 'vs/base/common/types';
+import { IAccessibilityService } from 'vs/platform/accessibility/common/accessibility';
+import { IContextKeyService } from 'vs/platform/contextkey/common/contextkey';
+import { TestCodeEditorService } from 'vs/editor/test/browser/editorTestServices';
+import { ICodeEditorService } from 'vs/editor/browser/services/codeEditorService';
+import { DefaultEndOfLine } from 'vs/editor/common/model';
+import { UndoRedoService } from 'vs/platform/undoRedo/common/undoRedoService';
+import { IThemeService } from 'vs/platform/theme/common/themeService';
+import { TestNotificationService } from 'vs/platform/notification/test/common/testNotificationService';
+import { TestDialogService } from 'vs/platform/dialogs/test/common/testDialogService';
+import { NotebookEditorStub, CellEditorProviderStub } from 'sql/workbench/contrib/notebook/test/stubs';
+import { QueryTextEditor } from 'sql/workbench/browser/modelComponents/queryTextEditor';
+import { NullTelemetryService } from 'vs/platform/telemetry/common/telemetryUtils';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { TestStorageService } from 'vs/workbench/test/common/workbenchTestServices';
+import { TestConfigurationService } from 'vs/platform/configuration/test/common/testConfigurationService';
+import { IEnvironmentService } from 'vs/platform/environment/common/environment';
+import { IStorageService } from 'vs/platform/storage/common/storage';
+
+class TestNotebookEditor extends NotebookEditorStub {
```

Comment:
It'd probably be useful to just move these test classes out into their own files so other tests can easily reuse them